tests/dummy/app/pods/button/template.hbs

```diff
@@ -7,533 +7,416 @@
 
 <div class='dummy-body button'>
 
-  <div class="section-title">Primary</div>
+  <div class='section-title'>Primary</div>
   <hr>
   <div class='section'>
-
     <div class='example'>
       <div class='title'>small</div>
       <div class='demo'>
+        {{! BEGIN-SNIPPET button-primary-small }}
         {{frost-button
           priority='primary'
           size='small'
           text='Text'
         }}
+        {{! END-SNIPPET }}
       </div>
       <div class='snippet'>
-        {{format-markdown "```handlebars
-{{frost-button
-  priority='primary'
-  size='small'
-  text='Text'
-}}
-```"}}
+        {{code-snippet name='button-primary-small.hbs'}}
```

Comment:
They use the extension to tell highlight.js how to deal with the code and there are two styles of snippet supported - inline and file based.  So they just load up all of the files and inline flags in memory - if the inline snippet occurs in an hbs file then it gets an hbs extension in the mapping